### ISSUE TITLE
Configuring 4G Module

### ISSUE THREAD

**Issue by kidmock** [bg111/asterisk-chan-dongle#276]
I have a SIM7600 module that I used primarily for SMS messages from my home automation and alarm system.

For my original use case it works great. However, I figure I should be able to use it for my home phone as well.

I can plug a headset in to the audio jack and use the AT commands to make and receive phone calls, no problem.

I stood up a quick and dirty asterisk box on a raspberry pi and configured chan_dongle. This doesn't seem to work.
chan_dongle sees:
  Voice                   : No

  SMS                     : Yes

Using chan_dongle as a trunk I can make calls, but have no audio in either direction. I can get an extension to ring, but again I have no audio in either direction.

It would seem possible that I should be able to make this work but can't seem to find information I'm looking for.

My question is does anyone know of a way in which I can use the audio jack in combination with a way to issue AT commands to this module.

**Comment by CyrusTheV** [comment]
Hi Kidmock.
Although I cannot answer your question, I became happy when I saw your question. Would you please let me know how you connect  SIM7600 to Asterisk and send a message.
Did you do this via chan_dongle?
Kindly, let me know if you agree with hiring someone who can do this together.

**Comment by kidmock** [comment]
I have the sim7600g-h 4g which is the International version. I'm Using a T-Mobile MNVO in the US that provides Voice and Text only.

I have the jumpers set to position C and set the power on boot by placing a jumper between 3.3V and GND.

I plug the module in the the USB port (not the USB to UART) and my linux hosts recognize/create the following devices

```
/dev/ttyUSB0  - Diagnostic Port
/dev/ttyUSB1  - NEMA GPS 
/dev/ttyUSB2 - Console port for AT Command Port
/dev/ttyUSB3 - Console port for AT Command Port
/dev/ttyUSB4 - Audio Port
/dev/cdc-wdm0 - Network Interface
```

I set the carrier provided APN through a serial terminal emulator (in my case minicom)

```
AT+CGDCONT=1,"IPV4V6","APN_1"
AT+CGDCONT=2,"IPV4V6","APN_2"
```

For my original use case to send text alerts from my security and and home automation system.
I would pipe it to /dev/ttyUSB3 which is something like:

```
echo  "AT+CMGS=\"+15551231234\""> /dev/ttyUSB3
echo "My Message\x1a" >  /dev/ttyUSB3
```

Since I began playing with asterisk, I configured chan_dongle as:

```
[dongle0]
data=/dev/ttyUSB2
audio=/dev/ttyUSB4
```

I've tried using different ports for audio none of them work

but sending SMS messages is as easy as:

```
asterisk -rx "dongle sms dongle0 +15551231234 My Message" 
```

**Comment by CyrusTheV** [comment]
Thanks a bunch for your response, 
Have you ever checked your NAT setting in Asterisk? One way or have audio or having no audio problems are almost related to misconfigured NAT setting.
by the way, Asterisk verbose can help you find the source of your problem. Asterisk -rvvv  .
Do you have any URL to  buy "sim7600g-h 4g"?
I'm eager to make this scenario working ASAP and would be glad if there is any way we can communicate over live chat apps (e.g. Whatsapp , Telegram)

**Comment by kidmock** [comment]
I'm not doing data so it shouldn't be a NAT issue. I would think this should work like ISDN or PRI. But it's been a really long time since I've done any telephony work.

Inbound calls report
```
[2021-03-14 21:22:04] WARNING[31637]: at_response.c:967 at_response_clcc: [dongle0] pbx started for device not voice capable
[2021-03-14 21:22:08] ERROR[31637]: at_response.c:473 at_response_error: [dongle0] AT^DDSETEX failed
```

You can find the module at all the usual places just search for SIM7600G-H:

https://www.aliexpress.com/item/4000432508896.html?spm=a2g0o.productlist.0.0.759e41652X0njw&algo_pvid=11cb6fd0-a419-41ec-92a4-0d488c26cea1&algo_expid=11cb6fd0-a419-41ec-92a4-0d488c26cea1-12&btsid=0bb0623b16157698209328448eca20&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_

https://www.amazon.com/dp/B08434KK72/ref=cm_sw_em_r_mt_dp_7KHCD8S00SKTRMAADGBH

**Comment by CyrusTheV** [comment]
is there any way for you to test the module with another OS, like windows or Arduino?

**Comment by CyrusTheV** [comment]
I will test your method with SIM800L and let you know, i did that before but Chan_Dongle couldn't connect to the module...but this time i will try another method like you USB...

**Comment by kidmock** [comment]
I'm completely confused ... Or maybe I didn't explain my issue well. 

I can use the SIM7600G-H module to make phone calls all day. This is completely OS independent as the functionality to issue AT commands is a product of a serial (TTY) connection. Prior to tying to use asterisk, I was just using it to send Text alerts from my home automation and security system. 

Ardunio, Windows, Mac, ESP32, Wise Terminal, PDP-11. It doesn't matter...

I can plug a headset with a 3.5mm TRRS connection into the module. Then use any terminal emulator to the USB or UART.

To dial out:
```
ATD +15551231234
```
To answer a call
```
ATA
```
To Hangup
```
ATH
```

To send a Text
```
AT+CMGS=+15551231234
My Message
CTRL-Z
```

This issue I have is specific to using the module with Asterisk to send and receive voice calls.

The chan_dongle add-on seems to work no problem with sending and receiving SMS. It can issue the dial out, answer, and hang up commands. It just doesn't seem to recognize the USB audio channel. 

So I'm curious if I can get chan_dongle(or something else) to recognize audio channel or if the 3.5mm jack can be used to bridge the gap.Through alsa or jackd or some other mechanism.

**Comment by rohitg76** [comment]
Hi kidmock:

I had taken a cursory look into Chan_dongle code a few years back. So, I am not an expert on it, but still offering my humble opinion since I feel you may be missing one key point.

To my knowledge and understanding, Chan_dongle is specifically coded for Huawei chipset based UMTS/HSPA USB modems. It complies with ISO8859 standard that is used for SMS, hence, SMS will likely work via Chan_dongle interfaces, even with other ISO8859-compliant hardware, but for voice to work with other than Huawei chipset based hw, it may not be just a matter of USB audio compliance but a larger difference in underlying audio/streaming/channel/rtp drivers that may likely be using the Huawei chip instruction set and driver interfaces or some hw-based incompatibilities. I would suggest enabling kernel debugging or use kadb to gain some insight on that front.

I had also gathered on these forums a few years back that Chan_dongle is a well written driver, but sensitive one, given it’s dependence on Huawei’s own chipsets that were used prior to Huawei moving to HiSilicon chipset based dongles. Which is why one would find that Chan_dongle works well with older models of Huawei dongles but not with some of the newer ones. That said, I have had some success with the newer Huawei modems with Chan_dongle as well, which I have shared in the past on these forums.

Summarily, I feel many of us have made good use of Chan_dongle all these years – there should be some concerted effort to expand and enhance Chan_dongle so that it can work with more general purpose GSM HATs and dongle, including support for VoLTE and 4G/5G hw. At present, I see largely only one contributor (wdoekes) enhancing the Chan_dongle code base. It requires more heads and hands to develop further. It is indeed a very powerful and useful FOSS addition.

You may want to refer this link for more on Chan_dongle, if not referred already - https://wiki.e1550.mobi/doku.php

Regards,
Rohit

**Comment by CyrusTheV** [comment]
Greeting,

Dear Rohit,
Thanks a million for your response,

Is there any way for you to help me in making chan_dongle compatible with Asterisk ( I mean make a phone call and send/receive SMS) ?

I am also looking to employ someone who has adequate knowledge to do this?
Kindly, please let me know if you know anyone.

Best regards,
Cyrus Dalili

**Comment by rohitg76** [comment]
Hi Cyrus:

Chan_dongle is coded to work with Asterisk! It already works well with it – you just need to use a compatible dongle.

I think lots of building blocks have got mixed up in this thread. Either we take it offline or may be clarify a bit – what the expectation is.

Components being discussed are,

1) Asterisk
2) Chan_dongle
3) Raspberry Pi
4) GSM HATs like SIM7600G-H or SIM800L
5) Huawei UMTS/HSPA Dongle (model that is compatible with Chan_dongle)

1+2+3+5 = ALREADY WORKS FOR SMS & VOICE VIA ASTERISK+CHAN_DONGLE (we make a UC product for small businesses that makes use of this with support for 2G/3G as well as non-VoLTE 4G dongles)

1+2+3+4 = REPORTEDLY WORKS FOR SMS VIA ASTERISK+CHAN_DONGLE (as reported by kidmock)

1+2+3+4 = REPORTEDLY DOES NOT WORK FOR VOICE VIA ASTERISK+CHAN_DONGLE (as reported by kidmock and my guess that it will likely require the Chan_dongle driver to be tweaked or perhaps rewritten to support different hardware HATs/dongles. My guess is this should not be too difficult to code given that kidmock has already figured that AT commands are able to get the voice going. A crude driver that just issues AT commands but via Chan_dongle could be a quick and dirty solution.)

So, which combination of 1 to 5 are you wanting to see working? May be also explain the use-case – because there are a lot of ready-made building blocks that offer a working solution with Asterisk depending on the application and need.

Regards,
Rohit

**Comment by CyrusTheV** [comment]
Thank you at first for your cooperation and time,

Actually, I am looking forward this solution ( 1+2+3+4 ), as you said, and I have no idea about writing a driver for voice via  ASTERISK+CHAN_DONGLE to work.
My use-case is to have a cheap GSM gateway, and because purchasing " Huawei UMTS/HSPA Dongle (model that is compatible with Chan_dongle)" is almost impossible to me, so I want to have a GSM gateway via  : 1+2+3+4

1) Asterisk
2) Chan_dongle
3) Raspberry Pi
4) GSM HATs like SIM800L

**Comment by rohitg76** [comment]
Cyrus, how many GSM SIMs do you want to be able to connect? While Chan_dongle can manage multiple dongles (google and check – you’ll be able to find instances where folks have connected ~20 dongles using a powered USB hub, or ~4 using the 4x USB ports on the Pi but with a 3A psu powering the Pi since each dongle consumes 500-700ma. This however may not be possible with a HAT-based approach given only 1x 40pin gpio and SBC accessories can bump up the price quite a bit and may not necessarily remain cheap.

Just curious – why is getting a Huawei dongle impossible for you? You can order it from AliExpress or lookup the list and try some of the recent models that are known to work. A dongle will cost you ~$25-50 each and a GSM HAT will also likely cost roughly the same, if not more.

Regards,
Rohit

**Comment by CyrusTheV** [comment]
Dear Rohit,
I want at least 50s one-port GSM gateway because I am going to place them in 50s different locations.
I live in Iran, and the "E303h-2" is the closest supported  model (according to http://asterisk-service.com/en_US/page/chan-dongle-modems) which I can find, I can send/receive SMS by using AT Commands and "Minicom", but when I try to use it with chan_dongle I got " timedout while waiting 'OK' in response to 'AT' " error.

other Huawei dongles are new ones like "E3372s-153" which regarding the mentioned link are not supported. As a consequence, I have decided to use sim card modules like "SIM800L" which I can buy them easily and at lower cost in my country.

**Comment by rohitg76** [comment]
Hi Cyrus:

Please refer these two threads..

https://github.com/bg111/asterisk-chan-dongle/issues/242 (discussion from back in 2017 when I had juts started off with chan_dongle)
https://github.com/bg111/asterisk-chan-dongle/issues/267 (around 3372h-153, you’ll find a mention of a recent E3131 that does work for both sms+voice with chan_dongle)

I think the 1-port GSM gateway is an interesting idea. We have a ready solution for it (pi+asterisk+gui+chan_dongle+sipsecurity+storage+wifirouter = all in one). You can reach me 1-1 at ***@***.*** if you wish to discuss.

Regards,
Rohit

**Comment by hichkak** [comment]
Hi Rohit,

Thanks a million for your response,

I am pleased to contact you 1-1 by your e-mail, but unfortunately this site has limited your email address (It shows : at ***@***.***).
You can reach me by my phone number : 00989194823543 on WhatsApp as well as my ID: @cyrusd on Telegram.

Best regards,
Cyrus

**Comment by yoozieck** [comment]
@rohitg76 
Hello I have the exactly same problem. I would like to use 4G modem, as despite calling features, I have some bots working on Raspberry Pi. Same as @kidmock I managed to make both way calls but without media through Asterisk + chan_dongle.
It would be really helpful to have this working, as this is the last part of my project :) Chan_dongle recognizes this modem as "Voice: no", but at the same time I can make calls with media with the terminal help (yet I cannot forward them via SIP).

Please help here.

**Comment by rohitg76** [comment]
Hi yoozieck:

I think we are getting caught up by wishful thinking versus what is valid use for chan_dongle in current form. Just use an dongle like E1750 or E3131 and it will work for you.

I have already shared – the issue in both cases is – using  a compatible dongle since chan_dongle is coded specifically around Huawei chips and not others. If you use a different chipset-based dongle or HAT, then the cha_dongle driver will need a re-write or some tweak. It is not a bug but a feature request that you may perhaps submit to the features forum.

I’m sorry but I don’t think I have anything more to add at this time.

Regards,
Rohit

**Comment by yoozieck** [comment]
Hello @rohitg76,
Thank you for answering. Of course I understand that chan_dongle was made for Huawei chips, but they're getting outdated each day. That is why I asked if this is possible to enhance modems list to more modern ones. I will write it on feature requests forum and I hope that such functionality will be added soon. Thank you again.

Regards,
yoozieck

**Comment by spi43984** [comment]
I see the benefit of the SIM7600as well as it is available and doesn't need any firmware modifications as many Huawei UMTS dongles need to support audio.

I played around a little bit with a SIM7600E and figured out that chan_dongle uses AT^DDSETEX to enable PCM on the ttyUSB interface as soon the audio call is started. SIM7600 expects AT^CPCMREG instead.

So there are two ways to solve this:
1) change chan_dongle and recompile it yourself - the command you'll find in [https://github.com/wdoekes/asterisk-chan-dongle/blob/master/at_command.h](url) 

2) wait for a firmware update for the SIM7600 which supports AT^DDSETEX as well

The former you can check out right now, the latter has the beauty that both UMTS dongles and SIM7600 would be supported.

BTW - connecting multiple SIM7600 with a rpi via USB should work - has anyone tried that yet? As the rpi has only one UART stacking them on GPIO pins might work if the're all additionally connected via USB but I haven't tried that yet.

**Comment by spi43984** [comment]
> BTW - connecting multiple SIM7600 with a rpi via USB should work - has anyone tried that yet? As the rpi has only one UART stacking them on GPIO pins might work if the're all additionally connected via USB but I haven't tried that yet.

I talked to Waveshare who I got my SIM7600E-H from - they confirmed that it is possible to connect two SIM7600 to one rpi.

**Comment by yoozieck** [comment]
Hello,
@rohitg76 is there a way to send AT+CPCMREG=1 command to the modem right after incoming call answer or outgoing call dialled?
And then AT+CPCMREG=0 right after hangup? I noticed this is necessary for SIM7600 to transfer audio properly.

**Comment by yoozieck** [comment]
@spi43984 but have you managed to make a call with audio using SIM7600? I tried today, and even after I send command `dongle cmd dongle0 AT+CPCMREG=1`, I was not able to get audio working.

**Comment by spi43984** [comment]
> @spi43984 but have you managed to make a call with audio using SIM7600? I tried today, and even after I send command `dongle cmd dongle0 AT+CPCMREG=1`, I was not able to get audio working.

Nope. I changed the source code a bit to adapt SIM7600 audio interface commands (AT+CPCMREG instead AT^DDSETEX) but the audio interface is still not detected in Asterisk.

I had a brief discussion with the manufacturer regarding firmware changes. They were interested initially to understand the issue but haven't heard anything back yet. So that's probably a dead end.

**Comment by 1iri** [comment]
Any updates on this? I have the same issue, I'm trying to use a SIM7600, but only SMS works with Asterisk and chan-dongle, voice is not detected.
I understand that chan-dongle is a Huawei modem module, but was wondering if we can adapt it for the SIM7600 or other modules.
Thanks!

**Comment by spi43984** [comment]
Nope. I haven't heard anything back from vendor. Tried to change the code myself but as I am not an expert om that it wasn't very fruitful ;-)

**Comment by IchthysMaranathaCopy** [comment]
This may be of help, chan_dongle adapted for the Quectel 4g module EC25, which may be using same at commands as simcom with the same qualcomm modem. Haven't tested it myself though.
https://github.com/t4rd15/asterisk-chan-quectel

**Comment by 1iri** [comment]
Was able to compile the asterisk-chan-quectel, and load the module, SMS works, but still when querying the module I get Voice : No
Also tried to change the AT^DDSETEX to AT+CPCMREG in at_command.h but still no good...
Any other suggestions are welcomed here :)

**Comment by IchthysMaranathaCopy** [comment]
Sorry it didn't work for you. For some modules it's necessary to disable gps and use that port (/dev/ttyUSB1) for volte audio -may not be applicable for simcom though. Perhaps these discussions might contain some nuggets of useful info on getting voice to work
https://forums.quectel.com/t/ec25-au-brazil-lte-not-sucess-call-over-lte-volte/2575
https://forum.gl-inet.com/t/gl-mifi-ec25-af-asterisk-voice-channel/7245

**Comment by IchthysMaranathaCopy** [comment]
Codec might also be an issue, EC25 uses ALC5616 by default (with option to set PCM, NAU8814 or TLV320AIC3104), SIM7600 uses NAU8810. I think only the SIM7600C has an option to use PCM (more here https://simcom.ee/documents/SIM7X00/SIM7X00_Audio_Application_Note_V1.00.pdf )

**Comment by wjswlgus123** [comment]
Is there any update for this topic?
I also need 4g dongle which support asterisk for voice.
If who can do this please add my skype: leechul27

**Comment by IchthysMaranathaCopy** [comment]
I was wrong about the codec issue as that only concerns the voice path on the hardware side using I2S/I2C. I've got the Sim7600 to work but it's a little unstable at the moment. I'll be publishing writeup and working code for 4g modules like EC25 and Sim7600 soon. 
https://github.com/wdoekes/asterisk-chan-dongle/issues/149#issuecomment-964041308

**Comment by wjswlgus123** [comment]
Hi @manojmat29 

the voice is going with volte? or use CS network?

Regards

**Comment by IchthysMaranathaCopy** [comment]
Yes, volte, mine is a 4g only operator. I don't think these modems will work differently for CS and volte anyway.

**Comment by IchthysMaranathaCopy** [comment]
Writeup and working code for Quectel here https://github.com/IchthysMaranatha/asterisk-chan-quectel
Still working on Simcom

**Comment by wjswlgus123** [comment]
@manojmat29 

Thank you for your share I had bought EC25-EUX this code can support this module?
When i receive it i will test and update you 

Thank you

**Comment by IchthysMaranatha** [comment]
You're welcome, yes I think it should work with with that. It is one of their newer modules, at the very least, it will still work with UAC using bridged alsa even if the audio serial port has been discontinued. I've tested with the EC25-E module.
Ref https://empa.com/empaiot/Quectel_Products_Portfolio_Overview_V2.6.pdf

EC25-EUX only has this doc in its application notes https://forums.quectel.com/uploads/short-url/79VE3yUZiWsTQ5SQ3GnmxyH3mnW.pdf instead of this one https://forums.quectel.com/uploads/short-url/xnztA07u1c4hilREu66LYIY6NGR.pdf. So, most probably you will need to use additional bridged alsa channel with this.

**Comment by yoozieck** [comment]
@manojmat29 wow! I will try that on SIMCOM 7600 and let you all know.

**Comment by IchthysMaranatha** [comment]
Still working on simcom means I'm working on it, code for simcom is not yet ready. Should be ready this week.

**Comment by yoozieck** [comment]
@IchthysMaranatha yeah, I overlooked that previous message and I thought that software is ready for both Simcom and EC25. I am waiting then for simcom version. Thank you for your work!

**Comment by IchthysMaranatha** [comment]
simcom was very tricky to get working but praise God! Was able to do it finally, updated code here https://github.com/IchthysMaranatha/asterisk-chan-quectel

**Comment by IchthysMaranatha** [comment]
Support has been added for Quectel UAC

**Comment by wjswlgus123** [comment]
> You're welcome, yes I think it should work with with that. It is one of their newer modules, at the very least, it will still work with UAC using bridged alsa even if the audio serial port has been discontinued. I've tested with the EC25-E module. Ref https://empa.com/empaiot/Quectel_Products_Portfolio_Overview_V2.6.pdf
> 
> EC25-EUX only has this doc in its application notes https://forums.quectel.com/uploads/short-url/79VE3yUZiWsTQ5SQ3GnmxyH3mnW.pdf instead of this one https://forums.quectel.com/uploads/short-url/xnztA07u1c4hilREu66LYIY6NGR.pdf. So, most probably you will need to use additional bridged alsa channel with this.

Hi @manojmat29 

I can find quectel and state is Free but can not make call.

ubuntu*CLI> quectel show devices
ID           Group State      RSSI Mode Submode Provider Name  Model      Firmware          IMEI             IMSI             Number        
quectel0     0     Free       23   0    0       xxxx xxxx     EC25       EC25EUXGAR08A05M1 8xxxx 4xxxxxxxxxx  Unknown   

   -- Called Quectel/quectel0/01012345678
    -- Quectel/quectel0-010000000c is making progress passing it to SIP/g1001-0000000d
       > 0x7feec0031cd0 -- Strict RTP switching to RTP target address 45.135.48.60:14422 as source
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3257][C-0000000e]: app_dial.c:1680 wait_for_answer: Unable to write frametype: 2
[Nov 26 14:39:16] WARNING[3129]: at_response.c:2138 at_response: [quectel0] Receive NO CARRIER

what is the problem?

can you help me?

**Comment by IchthysMaranatha** [comment]
First identify ports correctly with minicom and make a call with minicom. Are you getting proper DSCI urc with minicom when making a call? Please post screenshot of call using minicom.

**Comment by IchthysMaranatha** [comment]
Also, are you trying to make serial audio work or are you trying UAC method outlined here https://github.com/IchthysMaranatha/asterisk-chan-quectel/discussions/2 ?

**Comment by wjswlgus123** [comment]
I was use serial audio it seems can not make call out.
would you please let me know to to use minicom ?

**Comment by IchthysMaranatha** [comment]
You can use minicom, dterm or even putty with windows, I won't be able to to help you through that process if you're not familiar with using modems and a terminal, maybe this might help you https://www.thegeekstuff.com/2013/05/modem-at-command/ . Looking at your log, it doesn't look like a call is going through at all, it's running into 'no carrier'. Unload the module chan_quectel and place a call to that number. Do you hear ringback on the phone you're calling from? Before getting it to work with Asterisk, it must work on its own. The code will not grant any capability that it does not currently have.

**Comment by andyreynolds99** [comment]
> I was use serial audio it seems can not make call out. would you please let me know to to use minicom ?

Just go to simcom 7600 website and download manual for it:
https://www.waveshare.com/wiki/SIM7600G-H_4G_HAT
You will find in it some keys how minicom can be used. Nothing special is there, just at terminal. BTW, in my case, I don't see the commands i'm typing, but they work and print result (OK, ERROR, etc) (what is important).

> You can use minicom, dterm or even putty with windows, I won't be able to to help you through that process if you're not familiar with using modems and a terminal, maybe this might help you https://www.thegeekstuff.com/2013/05/modem-at-command/ . Looking at your log, it doesn't look like a call is going through at all, it's running into 'no carrier'. Unload the module chan_quectel and place a call to that number. Do you hear ringback on the phone you're calling from? Before getting it to work with Asterisk, it must work on its own. The code will not grant any capability that it does not currently have.

In my case I was able to make work only Tera Term. I just wanted to run examples from the mentioned hat's manual. I don't understand why putty didn't work...

**Comment by andyreynolds99** [comment]
BTW, Are the devices from original simcom and Chinese waveshare much different?

**Comment by rajattyagiin** [comment]
Hi. Does this still works
